tests/e2e/multi_vc_utils.go

```diff
@@ -0,0 +1,459 @@
+							MatchLabels: map[string]string{
+								"key": "app",
+							},
+						},
+						TopologyKey: "topology.kubernetes.io/zone",
+					},
+				},
+			},
+		}
+
+	}
+	if isParallelPodMgmtPolicy {
+		statefulset.Spec.PodManagementPolicy = appsv1.ParallelPodManagement
+	}
+	statefulset.Spec.Replicas = &replicas
+
+	framework.Logf("Creating statefulset")
+	CreateStatefulSet(namespace, statefulset, client)
+
+	framework.Logf("Wait for StatefulSet pods to be in up and running state")
+	fss.WaitForStatusReadyReplicas(client, statefulset, replicas)
+	gomega.Expect(fss.CheckMount(client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+	ssPodsBeforeScaleDown := fss.GetPodList(client, statefulset)
+	gomega.Expect(ssPodsBeforeScaleDown.Items).NotTo(gomega.BeEmpty(),
+		fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset.Name))
+	gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+		"Number of Pods in the statefulset should match with number of replicas")
+
+	return statefulset
+}
+
+/*setNodeAffinitiesForStatefulSet creates and returns nodeSelectorTerms for a statefulSet*/
+func setNodeAffinitiesForStatefulSet(allowedTopologies []v1.TopologySelectorLabelRequirement,
+	allowedTopologyLen int) []v1.NodeSelectorTerm {
+	var nodeSelectorRequirements []v1.NodeSelectorRequirement
+	var nodeSelectorTerms []v1.NodeSelectorTerm
+	rackTopology := allowedTopologies[len(allowedTopologies)-1]
+	if len(rackTopology.Key) == 32 {
+		var nodeSelectorTerm v1.NodeSelectorTerm
+		var nodeSelectorRequirement v1.NodeSelectorRequirement
+		nodeSelectorRequirement.Key = rackTopology.Key
+		nodeSelectorRequirement.Operator = "In"
+		for i := 0; i < allowedTopologyLen; i++ {
+			nodeSelectorRequirement.Values = append(nodeSelectorRequirement.Values, rackTopology.Values[i])
+			nodeSelectorTerm.MatchExpressions = append(nodeSelectorRequirements, nodeSelectorRequirement)
+		}
+		nodeSelectorTerms = append(nodeSelectorTerms, nodeSelectorTerm)
+	}
+	return nodeSelectorTerms
+}
+
+/* setSpecificAllowedTopology returns allowedTopology map with specific topology fields and values */
+func setSpecificAllowedTopology(allowedTopologies []v1.TopologySelectorLabelRequirement,
+	topkeyStartIndex int, startIndex int, endIndex int) []v1.TopologySelectorLabelRequirement {
+	var allowedTopologiesMap []v1.TopologySelectorLabelRequirement
+	specifiedAllowedTopology := v1.TopologySelectorLabelRequirement{
+		Key:    allowedTopologies[topkeyStartIndex].Key,
+		Values: allowedTopologies[topkeyStartIndex].Values[startIndex:endIndex],
+	}
+	allowedTopologiesMap = append(allowedTopologiesMap, specifiedAllowedTopology)
+
+	return allowedTopologiesMap
+}
+
+/*
+If we have multiple statefulsets and PVCs/PVs created on a given namespace and for performing
+cleanup of these multiple sts creation, deleteAllStatefulSetAndPVs is used
+*/
+func deleteAllStatefulSetAndPVs(c clientset.Interface, ns string) {
+	StatefulSetPoll := 10 * time.Second
```

Comment:
Doesn't scaleDownNDeleteStsDeploymentsInNamespace method do the same thing? We can add a parameter in the same function to delete pvcs and PV also.

Reply:
deleteAllStatefulSetAndPVs this util will perform cleanup when we have multiple statefulsets, standalone pods and deployment pod and will perform cleanup of multiple stale pvcs,pv entries

scaleDownNDeleteStsDeploymentsInNamespace - this util takes care of deleting only single statefulset etc. and doesn't meet the multi-vc requirement I m looking for.

As discussed with you, Going forward, we can use deleteAllStatefulSetAndPVs which is a single line of method call and quite generic and it will going to take care of cleaning entire stale entries left in the namespace

let me know if you think otherwise